### ISSUE TITLE
[Bug]: "allow upload and modification" option in shared link not working any more in 3.7.3, 3.7.4, 3.8.0 Client

### ISSUE THREAD

**Issue by tstreibl** [nextcloud/desktop#5482]
### ⚠️ Before submitting, please verify the following: ⚠️

- [X] This is a **bug**, not a question or a configuration issue.
- [X] This issue is **not** already reported on Github (I've searched it).
- [X] Nextcloud Server and Desktop Client are **up to date**. See [Server Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) and [Desktop Releases](https://nextcloud.com/install/#install-clients) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/)

### Bug description

When a (passwort protected) shared link is created with the 3.7.3 desctop client and the option "allow upload and modification" ("Hochladen und Bearbeiten erlaube) is aktivated, it's not possible to upload or modify.  The same link created with the web interface works well.

When you check such a client generated link with the web interface you see that the option "modify" ("Bearbeiten") is checked. A "normal" Web-Interfe generated link shows "upload and modify allwed" ("Hochladen und Bearbeiten erlaubt"). So the web interface already reflects that the link generated with the desctop client is malicious:
wrong:
![grafik](https://user-images.githubusercontent.com/32412085/221829683-42bc9e5e-6a76-47ca-9177-ffc054113f6d.png)
correct:
![grafik](https://user-images.githubusercontent.com/32412085/221829789-86d0ab42-aa84-4c94-9d90-70193a12146f.png)

### Steps to reproduce

1. use the desctop client to create as link
2. configure the link to allow "upload and modification"
3. test the link in a web browser:  no upload symbol available ( i didn't test if i can modify)

### Expected behavior

link should work according to link configuration, as this was the case with previous versions of desctop client

### Which files are affected by this bug

all files / folfers you share with such a link

### Operating system

Windows

### Which version of the operating system you are running.

Windows 11 and Windows Terminal Server

### Package

Appimage

### Nextcloud Server version

doesn't matter; tested with newest and older version

### Nextcloud Desktop Client version

3.7.3

### Is this bug present after an update or on a fresh install?

Updated from a minor version (ex. 3.4.2 to 3.4.4)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### Are you using an external user-backend?

- [ ] Default internal user-backend
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

**Comment by tstreibl** [comment]
Sorry - I just found this is a Duplicate of https://github.com/nextcloud/desktop/issues/5439.  It can be deleted when the additional information is not needed any more.

**Comment by tstreibl** [comment]
bug still exists in 3.7.4 client.

**Comment by tstreibl** [comment]
bug still exists in 3.8.0 client.  

Is it useful to report errors in this forum? Obviously non of the developers cares.

Since three versions basic functionality does not work any more (those functions were previously ok!):
- Impossible to create links that allow uploads
- impossible to configure the expiry date properly (from my point of view this is a mayour security issue) 

There are several related issues which were ignored the same way. Scrolling this forum once before a new release is done could avoid all this trouble. I can understand that it takes time to fix complicated issues. Those here don't seem to fall in this cathegory and should ideally show up during routine testing before a new release is done.

**Comment by sharcslife** [comment]
Don't get me wrong.
I really appreciate all the work that Nexcloud GmbH and all contributors put into this project.
But as a user and local admin, I am also furious about this.
This essential feature hase been missing in three releases for over two months now.

With bug-handling like this it gets harder and harder every day to defend the decision to have chosen Nextcloud as our file handling software.

**Comment by tstreibl** [comment]
Version 3.8.2 finally closes this issue.
The problem, that the expiration date can not be set is still open. See #5495

This is a toy and not a tool.
